### PR TITLE
Allow for an empty instance name to be used

### PR DESCRIPTION
**Pull request by CBeerta**
I would like the ability to replace the distribution specific configuration to get an installation that looks a bit more default. If that makes sense.

This Change allows for _instance_ to be set to '', which then allows the use of standard locations for configuration files and init scripts.

Not sure though if this is something that is in line with what the Cookbook originally wants to do.

Thanks!

### CODE REVIEW

**Comment by legal90** [comment]
Hm, it looks good! 
I think, that solution will help to migrate from previous major versions of `mysql` cookbook.

**Comment by jordant** [comment]
Glad to see this change, but there appears to be an issue with it not setting the initial root password when the instance name is empty. Running this on trusty ..

mysql_service '' do
  port '3306'
  version '5.5'
  initial_root_password 'changeme'
  action [:create, :start]
end

will result in the root password not being set.

**Comment by nkadel-skyhook** [comment]
I'd welcome this: having the "/etc/init.d/mysql" and other copies of init scripts or working directories lying arouund gets confusing, and the enforced use of /var/lib/mysql-[instance] instead of /var/lib/mysql/ for the mysql.sock breaks default perl scripts.

**Comment by CBeerta** [comment]
@jordant: Leaving the instance name empty doesn't work properly indeed. I wouldn't expect it to.

You'll have to define the service as:

```
  mysql_service 'default' do
    port '3306'
    version '5.5'
    instance ''
    initial_root_password 'changeme'
    action [:create, :start]
  end
```

File: .kitchen.yml [hunk 1]
```diff
@@ -112,6 +112,16 @@ suites:
       mysql:
        version: '5.7'
 
+  - name: service57-empty-instance
```
Comment (nkadel-skyhook) [inline .kitchen.yml]: So you want to support this feature only for MySQL 5.7 ?

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @CBeerta

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by gggeek** [comment]
+1 for allowing to manage the 'standard' instance.

All I need for the current project is one db service, with many databases which can all be created inside it (quite a common scenario for web apps imho).

Forcing a "non-standard" service name had the following drawbacks:
- teach devs not to use the 'usual' init service to start/stop the db
- set a global env var MYSQL_HOME, or all connections to 'localhost' would not work, because when you use 'localhost' mysql tries to connect via pipe, and the pipe name is different
- set the same env var in cronjobs (cronjobs by default get no env)
- set the name of the different pipe in 3 settings in php.ini, in 2 separate files => 6 settings!)

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

File: .kitchen.yml [hunk 2]
```diff
@@ -112,6 +112,16 @@ suites:
...
+    run_list:
+    - recipe[mysql_service_test::empty-instance]
+    includes: [
+      'centos-7.1'
```
Comment (nkadel-skyhook) [inline .kitchen.yml]: centos-7.2 is out and is the current default.

**Comment by chef-supermarket** [comment]
Hi. Your friendly Curry bot here. Just letting you know that all commit authors have become authorized to contribute. I have added the "Signed CLA" label to this issue so it can easily be found in the future.

**Comment by klyonrad** [comment]
So eh... does this mean that you reject this feature? With what reasons?

**Pull request closed without merging**